extensions/binding/org.eclipse.smarthome.binding.mqtt.generic/src/main/java/org/eclipse/smarthome/binding/mqtt/generic/handler/ChannelConfig.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.mqtt.generic.handler;
+
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.binding.mqtt.generic.internal.AbstractMqttThingValue;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.transform.TransformationException;
+import org.eclipse.smarthome.core.transform.TransformationService;
+import org.eclipse.smarthome.io.transport.mqtt.MqttBrokerConnection;
+import org.eclipse.smarthome.io.transport.mqtt.MqttException;
+import org.eclipse.smarthome.io.transport.mqtt.MqttMessageSubscriber;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Contains parsed channel configuration and runtime fields like the channel value.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class ChannelConfig implements MqttMessageSubscriber {
```

Comment:
Oh yes I remember, that was a proof of concept class. It works, but configuration holder classes should be pure DTOs. It should read like this: `MessageSubscriber extends ChannelConfig implements MqttMessageSubscriber`